controllers/category.js

```diff
@@ -0,0 +1,228 @@
+ *
+ * @apiParam {Object} [categoryImages]  CategoryImages of the product.
+ *
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *
+ * @apiUse CategoryNotFoundError
+ */
+
+exports.update = async (req, res) => {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      res.status(422).json({ errors: errors.array() });
+      return;
+    }
+    const updatedCartegory = await Category.findByIdAndUpdate(
+      req.params.id,
+      {
+        $set: req.body,
+      },
+      { new: true }
+    );
+    res.status(200).json(updatedCartegory);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+
+/**
+ * @api {get} /Category/:id Get User information .
+ * @apiVersion 0.2.0
+ * @apiName GetCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id Category unique ID.
+ *
+ * @apiSuccess {String} categoryName  CategoryName of the Product.
+ * @apiSuccess {String} sunCategories   subCategories of the Product.
+ *
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "CategoryName": "women Ethics",
+ *       "subCategories": " sarees"
+ *     }
+ *
+ * @apiError CategoryNotFound The id of the User was not found.
+ *
+ * @apiErrorExample Error-Response:
+ *     HTTP/1.1 404 Not Found
+ *     {
+ *       "error": "CategoryNotFound"
+ *     }
+ */
+
+exports.getOne = async (req, res) => {
+  try {
+    const categoryData = await Category.findById(req.params.id).select(
+      "categoryName subCategories categoryImages slug "
+    );
+    res.status(200).json(categoryData);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+
+exports.SearchByCondition = (req, res, next) => {
+  const searchData = req.query.categoryName;
+  Category.find({ categoryName: { $regex: searchData, $options: "$a" } }).then(
+    (data) => {
+      res.send(data);
+    }
+  );
+};
+
+/**
+ * @apiDefine CategoryNotFoundError
+ *
+ * @apiError CategoryNotFound The id of the User was not found.
+ *
+ * @apiErrorExample Error-Response:
+ *     HTTP/1.1 404 Not Found
+ *     {
+ *       "error": "CategoryNotFound"
+ *     }
+ */
+
+/**
+ * @api {get} /Category/:id Request Category information
+ * @apiName GetCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id product unique ID.
+ *
+ * @apiSuccess {String} CategoryName CategoryName of the Product.
+ * @apiSuccess {String} subcategories  subcategories of the User.
+ *
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "categoryName": "women Ethic",
+ *       "subCategories": "suites"
+ *     }
+ *
+ * @apiUse CategoryNotFoundError
+ */
+
+exports.getAll = async (req, res) => {
+  try {
+    const { page, perpage } = req.query;
+    const options = {
+      page: parseInt(page, 10) || 1,
+      limit: parseInt(perpage, 10) || 10,
+      sort: { createdAt: "desc" },
+    };
+    const category = await Category.paginate({}, options);

```

Comment:
@shivanipanwar247 you still didn't include total category and searching, sorting

Reply:
done mam ,in pagination  with getAll data total docs showing  and also added  search condition by categoryName